### ISSUE TITLE
Mongodb "--version" output changed in 4.3 or 4.4

### ISSUE THREAD

**Issue by mlev** [nodkz/mongodb-memory-server#487]
Version: 7.0.0-beta-46
Mongo version: v4.4.4
Node verson: 14.15.5

When using a pre-installed system binary and passing MONGOMS_SYSTEM_BINARY - the version isn't parsed. In the past this only caused a warning but new versions of the beta throw the error `Couldnt find an version from system binary output!` so my tests fail.

On my system the `version` command returns this format:
```
root@bc5c9e6eb7c6:/home/node# /usr/local/bin/mongod --version
db version v4.4.4
Build Info: {
    "version": "4.4.4",
    "gitVersion": "8db30a63db1a9d84bdcad0c83369623f708e0397",
    "openSSLVersion": "OpenSSL 1.1.0l  10 Sep 2019",
    "modules": [],
    "allocator": "tcmalloc",
    "environment": {
        "distmod": "debian92",
        "distarch": "x86_64",
        "target_arch": "x86_64"
    }
}
```

The problem seems to be this line:
https://github.com/nodkz/mongodb-memory-server/blob/e48976d34d838ab03376c4f578e19d84f3327a5e/packages/mongodb-memory-server-core/src/util/MongoBinary.ts#L94

For me this command `spawnSync(binaryPath, ['--version']).toString()` produces `[object Object]` which is why the regex fails to match the value.
To get the actual output I need to run `spawnSync(binaryPath, ['--version']).stdout.toString()` instead.

**Comment by hasezoey** [comment]
> For me this command spawnSync(binaryPath, ['--version']).toString() produces [object Object] which is why the regex fails to match the value.

this is because its probably logged wrong

---

it seems like for 4.3 or 4.4 the output of `--version` changed:

<details>
<summary>Command Output</summary>

```
$ ./mongod-x64-ubuntu-4.0.20 --version
db version v4.0.20
git version: e2416422da84a0b63cde2397d60b521758b56d1b
OpenSSL version: OpenSSL 1.1.1f  31 Mar 2020
allocator: tcmalloc
modules: none
build environment:
    distmod: ubuntu1804
    distarch: x86_64
    target_arch: x86_64

$ ./mongod-x64-ubuntu-4.2.10 --version
db version v4.2.10
git version: 88276238fa97b47c0ef14362b343c5317ecbd739
OpenSSL version: OpenSSL 1.1.1f  31 Mar 2020
allocator: tcmalloc
modules: none
build environment:
    distmod: ubuntu1804
    distarch: x86_64
    target_arch: x86_64

$ ./mongod-x64-ubuntu-4.4.1 --version
db version v4.4.1
Build Info: {
    "version": "4.4.1",
    "gitVersion": "ad91a93a5a31e175f5cbf8c69561e788bbc55ce1",
    "openSSLVersion": "OpenSSL 1.1.1f  31 Mar 2020",
    "modules": [],
    "allocator": "tcmalloc",
    "environment": {
        "distmod": "ubuntu2004",
        "distarch": "x86_64",
        "target_arch": "x86_64"
    }
}
```

</details>

**Comment by hasezoey** [comment]
actually, i cannot reproduce i, what system are you trying to run this on?

(i will still change to the suggested `.stdout`)

**Comment by mlev** [comment]
Yeah I'm a bit surprised I'm the first to see this. I'm running this in a docker image built from the official `node:14` image. This is the entirety of the docker build.

```
FROM node:14

ARG MONGO_RELEASE=mongodb-linux-x86_64-debian92-4.4.4
RUN wget --no-check-certificate https://fastdl.mongodb.org/linux/$MONGO_RELEASE.tgz
RUN tar --directory /usr/local/bin -xzvf $MONGO_RELEASE.tgz --strip-components=2 $MONGO_RELEASE/bin/mongod
RUN rm $MONGO_RELEASE.tgz
```

**Comment by hasezoey** [comment]
d8d6749f585afb76037a35956a7160e330ed28a3 implements the mentioned `.stdout`, will be in `7.0.0-beta.47` after the build finished

**Comment by mlev** [comment]
Great thanks!

**Comment by hasezoey** [comment]
closing, because included in `7.0.0-beta.47`
